modules/dop/providers/projectpipeline/service_impl.go

```diff
@@ -842,12 +835,24 @@ func (p *ProjectPipelineService) getPipelineSource(sourceID string) (pipelineSou
 	return source.PipelineSource, nil
 }
 
-func (p *ProjectPipelineService) batchGetPipelineDefinition(pipelineDefinitionIDArray []string) (map[string]*dpb.PipelineDefinition, error) {
-	var pipelineDefinitionListRequest dpb.PipelineDefinitionListRequest
-	pipelineDefinitionListRequest.IdList = pipelineDefinitionIDArray
-	pipelineDefinitionListRequest.PageNo = 1
-	pipelineDefinitionListRequest.PageSize = int64(len(pipelineDefinitionIDArray))
-	resp, err := p.PipelineDefinition.List(context.Background(), &pipelineDefinitionListRequest)
+func (p *ProjectPipelineService) batchGetPipelineDefinition(pipelineDefinitionIDArray []string, projectID uint64) (map[string]*dpb.PipelineDefinition, error) {
+	projectDto, err := p.bundle.GetProject(projectID)
+	if err != nil {
+		return nil, err
+	}
+	orgDto, err := p.bundle.GetOrg(projectDto.OrgID)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
should introduce a func like `getLocationByProjectID` or enhance `makeLocation` to support make location by projectID

Reply:
done